### PR TITLE
Migrating to swift 2.3

### PR DESCRIPTION
**Pull request by FranDepascuali**
## Summary ##

Migrates to Swift 2.3

### CODE REVIEW

**Comment by guidomb** [comment]
@danielaRiesgo could you take a look at this? @FranDepascuali Thanks for the PR!

File: Core/Extensions/Foundation/NSURL.swift [hunk 1]
```diff
@@ -16,7 +16,7 @@ import Foundation
  
  - returns: The new NSURL, with pathComponent appended to baseURL
  */
-public func / (baseURL: NSURL, pathComponent: String) -> NSURL {
+public func / (baseURL: NSURL, pathComponent: String) -> NSURL? {
```
Comment (danielaRiesgo) [inline Core/Extensions/Foundation/NSURL.swift]: Why could this be None?

File: Core/Extensions/Foundation/NSURL.swift [hunk 2]
```diff
@@ -28,7 +28,7 @@ public func / (baseURL: NSURL, pathComponent: String) -> NSURL {
  
  - returns: The new NSURL, with pathComponent appended to baseURL
  */
-public func / (baseURL: NSURL, pathComponent: Int) -> NSURL {
+public func / (baseURL: NSURL, pathComponent: Int) -> NSURL? {
```
Comment (danielaRiesgo) [inline Core/Extensions/Foundation/NSURL.swift]: Same here.

File: Core/Extensions/Foundation/NSURL.swift (hunk 1 above)
Comment (FranDepascuali) [inline Core/Extensions/Foundation/NSURL.swift]: The signature of this function was changed by apple
screen shot 2016-11-12 at 15 29 39

File: Core/Extensions/Foundation/NSURL.swift (hunk 2 above)
Comment (FranDepascuali) [inline Core/Extensions/Foundation/NSURL.swift]: Same as above

**Comment by FranDepascuali** [comment]
@guidomb @danielaRiesgo I don't know why travis is failing. It's something with carthage cache. I don't have permissions to run the tests again, neither to merge any PR, so I will be just submitting PR's. 
(I'm glad to be working with you two again :D).

BTW, I'm using wolmo core in https://github.com/FranDepascuali/ProyectoAlimentar and it's helping a lot 👍 .

**Comment by danielaRiesgo** [comment]
Maybe we should change the xcode-version to 8 in `.travis.yml` file?

**Comment by gjcairo** [comment]
@guidomb @danielaRiesgo should we close this, given I started #29?

**Comment by guidomb** [comment]
Yes, we won't support Swift 2.3.

**Pull request closed without merging**